### ISSUE TITLE
Problem importing module strings.py: cannot import name 'Counter'

### ISSUE THREAD

**Issue by pkolbus** [PyCQA/pylint#4412]
On python 3.6.0, usage of pylint 2.8.2 results in an import warning to stdout.  

N.B. `typing.Counter` was added in python 3.6.1, so this import probably needs to be wrapped in a `TYPE_CHECKING` guard.

(Yes, I know 3.6.0 is an old patch release. Still have it on some CI servers for unfortunate reasons.)

### Steps to reproduce

Given that this warning seems to be coming from `register_plugins()` in utils.py, it is probably reproducible with any input file.

### Current behavior

Printed message: `Problem importing module strings.py: cannot import name 'Counter'`

### Expected behavior

No warning, and `strings` checker works as expected.

### pylint --version output

Result of `pylint --version` output:

```
pylint 2.8.2
astroid 2.5.6
Python 3.6.0 (default, Jan 23 2017, 20:01:14) [MSC v.1900 64 bit (AMD64)]
```

**Comment by pkolbus** [comment]
Happy to contribute a patch if you like.

**Comment by cdce8p** [comment]
> Happy to contribute a patch if you like.

Go for it! Tag me and I'll make sure to review it.

Had to search a bit, but it seams as this commit is the issue: [link](https://github.com/PyCQA/pylint/commit/9da9c6c5e9c7005bf6c26bb6be690e7b684d69ce#diff-d68f9eb5792cdb1741008f54fd29779c9e1f0b22584813c41c81cd5d1034d687)
